opal/mca/btl/uct/btl_uct_component.c

```diff
@@ -124,6 +135,54 @@ static void mca_btl_uct_mem_release_cb(void *buf, size_t length, void *cbdata, b
     ucm_vm_munmap(buf, length);
 }
 
+static void mca_btl_uct_component_parse_include_list (const char *value, mca_btl_uct_include_list_t *list) {
+    list->list = NULL;
+    list->include = true;
+
+    if (value == NULL) {
+        return;
+    }
+
+    if (value[0] == '^') {
+        list->include = false;
+        value++;
+    }
+
+    list->list = opal_argv_split(value, ',');
+}
+
+static void mca_btl_uct_include_list_free (mca_btl_uct_include_list_t *list) {
+    opal_argv_free (list->list);
+    list->list = NULL;
+}
+
+int mca_btl_uct_include_list_rank (const char *name, const mca_btl_uct_include_list_t *list) {
+    if (list->list == NULL) {
+        return -1;
+    }
+
+    for (int i = 0; list->list[i]; ++i) {
+        regex_t preg;
```

Comment:
Why regex ? You compare an official (MD or TL) name to the list of include, which should only contains correct names for these objects. Is this only for case insensitivity ?

Reply:
It is probably overkill but I wanted to make the matching more powerful. Originally, when I first started working with these systems the names were based on where (pci bus number, etc) the HCA was connected. Example: rocep0s4 (now simplified to irdma0 but kept the regex matching). Regex is a simple way to get explicit partial matching (using .*).